### PR TITLE
Marking an allocation chained if we are creating this to replace an o…

### PR DESCRIPTION
**Pull request by diptanu**
…ld one

### CODE REVIEW

File: scheduler/generic_sched_test.go [hunk 1]
```diff
@@ -2054,3 +2054,85 @@ func TestGenericSched_FilterCompleteAllocs(t *testing.T) {
...
+	var allocIDs []string
+	for _, allocList := range h.Plans[0].NodeAllocation {
+		for _, alloc := range allocList {
+			allocIDs = append(allocIDs, alloc.ID)
+		}
+	}
+	sort.Strings(allocIDs)
+
+	// Create a new harness to invoke the scheduler again
+	h1 := NewHarnessWithState(t, h.State)
```
Comment (dadgar) [inline scheduler/generic_sched_test.go]: You should be able to just use the same harness?

File: scheduler/generic_sched.go [hunk 2]
```diff
@@ -457,6 +457,12 @@ func (s *GenericScheduler) computePlacements(place []allocTuple) error {
 				ClientStatus:  structs.AllocClientStatusPending,
 			}
 
+			// If the new allocation is replacing an older allocation then we
+			// set the record the older allocation id so that they are chained
+			if missing.Alloc != nil {
```
Comment (dadgar) [inline scheduler/generic_sched.go]: System scheduler

File: scheduler/generic_sched_test.go (hunk 1 above)
Comment (diptanu) [inline scheduler/generic_sched_test.go]: We need a new test harness here with the existing state so that the data structures of the harness are empty before the planner is invoked again.

**Comment by dadgar** [comment]
LGTM
